libraries/llvm/effekt.effekt

```diff
@@ -1,38 +1,57 @@
 module effekt
 
 extern include "rts.ll"
+extern include "primitives.ll"
 
 
-// I/O primitives
-// ==============
+// i/o
 
 extern effect Console
-extern include "primitives.ll"
 
-extern pure def println(n: Int): Unit / Console =
-  "call void @c_println_int64(i64 %n)   ret %Pos zeroinitializer"
+extern io def println(n: Int): Unit / Console =
+  "%unit = call %Pos @c_println_Int(%Int %n)   ret %Pos %unit"
 
-extern pure def println(b: Boolean): Unit / Console =
-  "call void @c_println_pos_boolean(%Pos %b)   ret %Pos zeroinitializer"
+extern io def println(b: Boolean): Unit / Console =
+  "%unit = call %Pos @c_println_Boolean(%Pos %b)   ret %Pos %unit"
 
+extern io def println(x: Double): Unit / Console =
+  "%unit = call %Pos @c_println_Double(%Double %x)   ret %Pos %unit"
 
-// integer primitives
-// ==================
+
+// integer arithmetic
 
 extern pure def infixAdd(x: Int, y: Int): Int =
-  "%z = add %Int %x, %y ret %Int %z"
+  "%z = call %Int @c_add_Int_Int(%Int %x, %Int %y)   ret %Int %z"
+
+extern pure def infixSub(x: Int, y: Int): Int =
+  "%z = call %Int @c_sub_Int_Int(%Int %x, %Int %y)   ret %Int %z"
 
 extern pure def infixMul(x: Int, y: Int): Int =
-  "%z = mul %Int %x, %y ret %Int %z"
+  "%z = call %Int @c_mul_Int_Int(%Int %x, %Int %y)   ret %Int %z"
```

Comment:
Does this have any implications on performance?

Reply:
It very well may, since the linker only has a vague understanding on inlineability; I reverted in https://github.com/effekt-lang/effekt/pull/141/commits/6f1ed0204e7086de8607d66b046108ac2e024340, though.